From 99e57a702164c23899b897cbf56d890f12b1ee9f Mon Sep 17 00:00:00 2001
Message-ID: <99e57a702164c23899b897cbf56d890f12b1ee9f.1741637131.git.tavianator@tavianator.com>
From: Andre Klapper <a9016009@gmx.de>
Date: Fri, 30 Aug 2024 16:37:01 +0200
Subject: [PATCH] Fix implicitly nullable parameter declarations for PHP 8.4

Summary:
Implicitly nullable parameter declarations are deprecated in PHP 8.4:
https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated

The proposed syntax was introduced in PHP 7.1.

Refs T15935

Test Plan: Try to successfully run `./bin/storage upgrade` with PHP 8.4

Reviewers: O1 Blessed Committers, chris

Reviewed By: O1 Blessed Committers, chris

Subscribers: tobiaswiese, valerio.bozzolan, Matthew, Cigaryno

Maniphest Tasks: T15935

Differential Revision: https://we.phorge.it/D25813
---
 src/utils/utils.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/utils.php b/src/utils/utils.php
index e7b6d66d..fe9f0032 100644
--- a/src/utils/utils.php
+++ b/src/utils/utils.php
@@ -960,7 +960,7 @@ function phutil_split_lines($corpus, $retain_endings = true) {
  * @param   list  $list (optional) List of scalars.
  * @return  dict  Dictionary with inputs mapped to themselves.
  */
-function array_fuse(array $list = null) {
+function array_fuse(?array $list = null) {
   if ($list) {
     return array_combine($list, $list);
   }
-- 
2.48.1

